examples/org.eclipse.xtext.xtext.ui.examples/contents/org.eclipse.xtext.example.homeautomation/src/org/eclipse/xtext/example/homeautomation/formatting2/RuleEngineFormatter.xtend

```diff
@@ -107,170 +91,55 @@ class RuleEngineFormatter extends XbaseFormatter {
-		expr.regionForKeyword(")").prepend[noSpace]
-		expr.initExpressions.forEach[it.format(format)]
-		expr.expression.prepend[oneSpace]
-		expr.expression.format(format)
-		expr.updateExpressions.head.prepend[oneSpace]
-		expr.updateExpressions.forEach[it.format(format)]
-		if (!(expr.eachExpression instanceof XBlockExpression)) {
-			expr.eachExpression.prepend[newLine; increaseIndentation].append[decreaseIndentation]
-		}
-		expr.eachExpression.format(format)
-	}
-	
-	override dispatch void format(XSynchronizedExpression expr, extension IFormattableDocument format) {
-		expr.param.surround[noSpace]
-		expr.param.format(format)
-		if (!(expr.expression instanceof XBlockExpression)) {
-			expr.expression.prepend[newLine; increaseIndentation].append[decreaseIndentation]
-		}
-		expr.expression.format(format)
+
+	override protected void formatBody(XExpression expr, boolean forceMultiline, extension IFormattableDocument doc) {
+		if (expr == null)
+			return;
+		if (expr instanceof XBlockExpression) {
+			expr.prepend[newLine]
+		} else if (forceMultiline || expr.leadingHiddenRegion.isMultiline) {
+			expr.prepend[newLine].surround[indent]
+		} else {
+			expr.prepend[oneSpace]
+		}
+		expr.format(doc)
 	}
-	
-	override dispatch void format(XTryCatchFinallyExpression expr, extension IFormattableDocument format) {
-		if (!(expr.expression instanceof XBlockExpression)) {
-			expr.expression.prepend[newLine; increaseIndentation].append[decreaseIndentation]
-		}
-		expr.expression.format(format)
-		for (cc : expr.catchClauses) {
-			if (!(cc.expression instanceof XBlockExpression)) {
-				cc.expression.prepend[newLine; increaseIndentation]
-				if (cc != expr.catchClauses.last || expr.finallyExpression != null)
-					cc.expression.append[newLine; decreaseIndentation]
-				else
-					cc.expression.append[decreaseIndentation]
-			}
-			cc.format(format)
-		}
-		if (expr.finallyExpression != null) {
-			if (!(expr.finallyExpression instanceof XBlockExpression)) {
-				expr.finallyExpression.prepend[newLine; increaseIndentation].append[decreaseIndentation]
-			}
-			expr.finallyExpression.format(format)
-		}
+
+	override protected void formatBodyInline(XExpression expr, boolean forceMultiline,
+		extension IFormattableDocument doc) {
+		if (expr == null)
+			return;
+		if (expr instanceof XBlockExpression) {
+			expr.surround[newLine]
+		} else if (forceMultiline || expr.leadingHiddenRegion.isMultiline) {
+			expr.prepend[newLine].surround[indent].append[newLine]
+		} else {
+			expr.surround[oneSpace]
+		}
+		expr.format(doc)
 	}
-	
-	override dispatch void format(XCatchClause expr, extension IFormattableDocument format) {
-		expr.regionForKeyword("catch").append[oneSpace]
-		expr.declaredParam.surround[noSpace]
-		expr.declaredParam.format(format)
-		expr.expression.format(format)
+
+	override protected void formatBodyParargraph(XExpression expr, extension IFormattableDocument doc) {
```

Comment:
fixed in d9399cc1416b1ad454e0bc055a368eba13d418b6 and 8b08a7b2cf62f63856e487fad3ca2366af782363
On 15 Apr 2015, at 16:07, Sebastian Zarnekow <notifications@github.com> wrote:

> In examples/org.eclipse.xtext.xtext.ui.examples/contents/org.eclipse.xtext.example.homeautomation/src/org/eclipse/xtext/example/homeautomation/formatting2/RuleEngineFormatter.xtend:
> 
> >  	}
> > -	
> > -	override dispatch void format(XCatchClause expr, extension IFormattableDocument format) {
> > -		expr.regionForKeyword("catch").append[oneSpace]
> > -		expr.declaredParam.surround[noSpace]
> > -		expr.declaredParam.format(format)
> > -		expr.expression.format(format)
> > +
> > +	override protected void formatBodyParargraph(XExpression expr, extension IFormattableDocument doc) {
> Typo
> 
> —
> Reply to this email directly or view it on GitHub.
>